source/common/router/config_impl.h

```diff
@@ -775,7 +775,10 @@ class RouteEntryImplBase : public RouteEntryAndRoute,
   std::string newUri(const Http::RequestHeaderMap& headers) const override;
   void rewritePathHeader(Http::RequestHeaderMap&, bool) const override {}
   Http::Code responseCode() const override { return direct_response_code_.value(); }
-  const std::string& responseBody() const override { return direct_response_body_; }
+  const std::string& responseBody() const override {
+    absl::MutexLock lock(&direct_response_mutex_);
+    return direct_response_body_;
```

Comment:
There still maybe a contention because you return a reference to the body string? May be TLS should be used here to ensure the thread-safe.

Reply:
Can you please expand more on this? How should I use TLS here?